numpy/core/overrides.py

```diff
@@ -34,6 +35,15 @@ def set_array_function_like_doc(public_api):
     return public_api
 
 
+def add_signature(function, docstring):
+    try:
+        function.__signature__ = inspect._signature_fromstr(
```

Comment:
Would indeed be nice to replace this with a vendored copy or something else that's not private API.